packages/xod-project/src/typeDeduction.js

```diff
@@ -192,26 +193,62 @@ const omitIntermediateAmbiguousPins = (
     )
   );
 
+// We use `toSet` and `inSet` in `deducePinTypes`
+// cause it is faster than `isAmong` function.
+// :: [a] -> Set [a]
+const toSet = a => new Set(a);
+// :: a -> Set [a] -> Boolean
```

Comment:
I think the signature `Set [a]` is wrong. Should it be `Set a`?

And I think it’s a better idea to move the usage note comment to the place of actual usage. Imagine you’d promote the two funcs to `xod-func-tools`.